src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -721,6 +721,8 @@ class JavaThread: public Thread {
   bool VTMS_transition_mark() const              { return Atomic::load(&_VTMS_transition_mark); }
   void set_VTMS_transition_mark(bool val)        { Atomic::store(&_VTMS_transition_mark, val); }
 
+  bool hide_jvmti_events() const                 { return _is_in_VTMS_transition || _is_disable_suspend; }
```

Comment:
The name
hide_jvmti_events()
looks incorrect, like we change thread state to hide jvmti event and fail/return if can't.
I prefer something like
jvmti_events_hidden()
is_jvmti_event_posting_hidden()
or 
should_hide_jvmti_events()
To make clear that we read only some state.
Also, agree with Chris that comments are needed only in the implementation of this method.

Good to know when thread can and can't post events.

Reply:
Thank you for the comment.
I like this suggestion: `should_hide_jvmti_events()`.